arraycontext/container/traversal.py

```diff
@@ -842,7 +845,7 @@ def _to_numpy_with_check(subary: Any) -> Any:
                     f"array of type '{type(subary).__name__}' not in "
                     f"supported types {actx.array_types}")
 
-    return rec_map_array_container(_to_numpy_with_check, ary)
+    return rec_map_array_container(_to_numpy_with_check, freeze(ary, actx))
```

Comment:
Is this correct? (Calling `freeze` for `to_numpy`?)

Reply:
Yep, was incorrect. Replaced with `thaw(freeze(...))`. See 98776255f9cf518172bc57a35d75380f948e6ef0.